controllers/integration_test.go

```diff
@@ -146,6 +146,23 @@ var _ = Describe("Integration Test", func() {
 		}
 	} // advanceStateAndCheckReady(state dwsv1alpha1.WorkflowState, w *dwsv1alpha1.Workflow)
 
+	checkPSIConsumerReference := func(storageName string, w *dwsv1alpha1.Workflow) {
+		persistentInstance = &dwsv1alpha1.PersistentStorageInstance{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      storageName,
+				Namespace: w.Namespace,
+			},
+		}
+
+		By(fmt.Sprintf("Retrieving PSI '%s'", storageName))
+		Expect(k8sClient.Get(context.TODO(), client.ObjectKeyFromObject(persistentInstance), persistentInstance)).To(Succeed(), "PersistentStorageInstance created")
+
+		By("Checking PSI has the correct consumer reference")
+		Expect(len(persistentInstance.Spec.ConsumerReferences)).To(Equal(1))
```

Comment:
There's a `HaveLen()` matcher for checking arrays
`Expect(persistentInstance.Spec.ConsumerReferences).To(HaveLen(1))`

Reply:
Nice. I'll use that.